### PR TITLE
[vcpkg] update fetch command

### PR DESCRIPTION
**Pull request by dan-shaw**

### CODE REVIEW

File: toolsrc/src/vcpkg/commands.fetch.cpp [hunk 1]
```diff
@@ -18,6 +20,36 @@ namespace vcpkg::Commands::Fetch
...
+            const std::string repo = "https://github.com/microsoft/vcpkg";
+            fs::path fetch_dir = paths.root / fs::u8path("vcpkg-fetch");
+
+            Files::Filesystem& fs = paths.get_filesystem();
+
+            fs.remove_all(fetch_dir, ignore_errors);
+            fs.create_directory(fetch_dir, VCPKG_LINE_INFO);
+
+            const std::string full_cmd = Strings::format(
+                R"(%s clone %s %s)", fs::u8string(git_exe), fs::u8string(repo), fs::u8string(fetch_dir));
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/commands.fetch.cpp:36]:
```suggestion
                R"("%s" clone "%s" "%s")", fs::u8string(git_exe), fs::u8string(repo), fs::u8string(fetch_dir));
```

This needs to be made resistant to spaces in the path

File: toolsrc/src/vcpkg/commands.fetch.cpp [hunk 2]
```diff
@@ -18,6 +20,36 @@ namespace vcpkg::Commands::Fetch
...
+
+            auto output = System::cmd_execute_and_capture_output(full_cmd);
+
+            std::error_code ec;
+            fs.remove_all(paths.ports, ignore_errors);
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/commands.fetch.cpp:41]: I don't think ignore errors is the right behavior here. We want failures to be loud so we don't get "impossible" bugs where only some part of the tree got updated or similar.

It's also very strange to totally ignore these errors but then explicitly check for the git exit code on 48.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
* If this is in support of the VS Code extension, why doesn't the VS Code extension do this rather than the vcpkg tool? VS Code should already know how to do git clones etc. which might save a download here.
* What frequency do we expect the VS Code extension to use when invoking this?
* I'm concerned that once a user touches this their update experience, which we've previously documented as `git pull`, becomes broken.
* This is really a totally different 'animal' to the other `vcpkg fetch` invocations, it seems like it should be a top level command and accept the path as a parameter rather than just trying to use the root?

File: toolsrc/src/vcpkg/commands.fetch.cpp (hunk 1 above)
Comment (strega-nil) [reply toolsrc/src/vcpkg/commands.fetch.cpp:36]: I believe we have an "escape for command line" function.

**Comment by BillyONeal** [comment]
@dan-shaw Should we close this now that registries have landed?

**Comment by BillyONeal** [comment]
I believe this PR can be closed following https://github.com/microsoft/vcpkg/pull/15474/

**Comment by JackBoosY** [comment]
According to @BillyONeal said, close this PR.
If you have any concern, please reopen this.

**Pull request closed without merging**